### PR TITLE
feat(launchpad): create config file from GUI

### PR DESCRIPTION
**Pull request by lmiller1990**
Creates a `cypress.config` file from the GUI based on the framework, bundler and language of choice.

@tgriesser is this how you planned for `actions` to be used? 

This PR makes a few other tiny "quality of life changes", like using `defineProps` instead of `props` for better IDE integration. I'm moving components to use `defineProps` as I see them, since then we can use regular TS interfaces instead of the object syntax.

### CODE REVIEW

File: packages/graphql/src/entities/Project.ts [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
+  isCurrent: boolean
   projectRoot: string
   projectBase: ProjectBaseContract
 }
 
 @nxs.objectType({
   description: 'A Cypress project is a container',
 })
-export class Project {
+export class Project implements ProjectBaseContract {
```
Comment (lmiller1990) [inline packages/graphql/src/entities/Project.ts:14]: This should be implementing `ProjectBaseContract` right?

File: packages/graphql/test/entities/Wizard.spec.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { expect } from 'chai'
```
Comment (lmiller1990) [inline packages/graphql/test/entities/Wizard.spec.ts:1]: Not sure if we want to test these in isolation or in a more e2e fashion by spawning a gql server. Either way, we have a navigation bug so it's good to add some tests to make sure it doesn't regress. Any recommendation for testing the gql logic @tgriesser?

File: packages/launchpad/src/components/ConfigFile.vue [hunk 3]
```diff
@@ -59,51 +65,72 @@ fragment ConfigFile on Wizard {
...
+
+import("prismjs/components/prism-typescript").then(() => {
+  tsInstalled.value = true;
 });
+
+
+const createConfigFile = useMutation(AppCreateConfigFileDocument)
+
+const createConfig = async () => {
+  console.log('Creating!')
```
Comment (JessicaSachs) [inline packages/launchpad/src/components/ConfigFile.vue:112]:
```suggestion
```

File: packages/launchpad/src/components/ConfigFile.vue [hunk 4]
```diff
@@ -59,51 +65,72 @@ fragment ConfigFile on Wizard {
...
-    };
-  },
+const altFn = (val: boolean) => {
+  manualInstall.value = val
+}
+
+const tsInstalled = ref(false);
+const language = ref<"js" | "ts">("ts");
+const nextButtonName = computed(() =>
+  manualInstall.value ? "I've added this file" : "Create File"
```
Comment (JessicaSachs) [inline packages/launchpad/src/components/ConfigFile.vue:101]: Before you merge, rebase on the i18n PR and then fix this

File: packages/launchpad/src/components/WizardLayout.vue [hunk 5]
```diff
@@ -39,41 +38,26 @@ mutation WizardLayoutNavigate($direction: WizardNavigateDirection!) {
...
+  defineProps<{
+    next: string
+    back: string
+    alt?: string
+    altFn?: (val: boolean) => void
+    nextFn?: (...args: unknown[]) => any
+}>(), {
+  next: 'Next Step',
+  back: 'Back Step',
+})
```
Comment (JessicaSachs) [inline packages/launchpad/src/components/WizardLayout.vue:51]: I almost did this on my side too lol. Nice.

File: packages/launchpad/src/components/ConfigFile.vue [hunk 6]
```diff
@@ -59,51 +65,72 @@ fragment ConfigFile on Wizard {
...
+  tsInstalled.value = true;
 });
+
+
+const createConfigFile = useMutation(AppCreateConfigFileDocument)
+
+const createConfig = async () => {
+  console.log('Creating!')
+  if (!props.app?.activeProject?.projectRoot) {
+    throw Error(`Expected active project to be set with a projectRoot. This should never happen.`)
```
Comment (JessicaSachs) [inline packages/launchpad/src/components/ConfigFile.vue:114]:
```suggestion
    throw Error(`Cannot find the active project's projectRoot. This should never happen.`)
```

File: packages/graphql/test/entities/Wizard.spec.ts (hunk 2 above)
Comment (JessicaSachs) [reply packages/graphql/test/entities/Wizard.spec.ts:1]: Yeah, I'd like advice on how to test this well.

File: packages/launchpad/src/components/WizardLayout.vue [hunk 7]
```diff
@@ -39,41 +38,26 @@ mutation WizardLayoutNavigate($direction: WizardNavigateDirection!) {
...
+const props = withDefaults(
+  defineProps<{
+    next: string
+    back: string
+    alt?: string
+    altFn?: (val: boolean) => void
+    nextFn?: (...args: unknown[]) => any
+}>(), {
+  next: 'Next Step',
+  back: 'Back Step',
```
Comment (JessicaSachs) [inline packages/launchpad/src/components/WizardLayout.vue:50]: You accidentally added a word.
```suggestion
  back: 'Back',
```

File: packages/launchpad/src/components/ConfigFile.vue [hunk 8]
```diff
@@ -59,51 +65,72 @@ fragment ConfigFile on Wizard {
...
+    configFilename: `cypress.config.${language.value}`
+  })
+}
+
+const code = computed(() => {
+  if (language.value === 'js') {
+    return props.wizard.sampleCodeJs
+  }
+  return props.wizard.sampleCodeTs
+})
```
Comment (JessicaSachs) [inline packages/launchpad/src/components/ConfigFile.vue:133]: Can you put this above the usage? I was struggling to figure out why this wasn't undefined.

**Review by JessicaSachs — APPROVED** [review]
The front-end looks good. I QA'd everything manually. A summary:
* small typos I've batched into suggestions
* there's a bug in the existing codegen for JavaScript. It doesn't generate valid JS.
* I'm curious about how to test the gql server. Aside from manually testing it how can I trust that it's working correctly?

**Comment by lmiller1990** [comment]
@JessicaSachs I am also not clear on the best practice for testing the gql server, I will look into this more. Let's hold of merging until we have some more tests in place.

**Pull request merged**